src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -601,59 +615,37 @@ export class MeetingManager implements AudioVideoObserver {
         innerDevice = device;
       }
 
-      if (innerDevice === null) {
-        this.selectedAudioInputDevice = null;
-      } else if (typeof innerDevice === 'string') {
-        this.selectedAudioInputDevice = innerDevice;
-      } else if (innerDevice instanceof MediaStream) {
-        this.selectedAudioInputDevice = innerDevice.id;
-      } else {
-        const deviceId = this.getDeviceID(innerDevice);
+      const deviceId =
+        DefaultDeviceController.getIntrinsicDeviceId(innerDevice);
+      if (typeof deviceId === 'string') {
         this.selectedAudioInputDevice = deviceId;
+      } else if (Array.isArray(deviceId) && deviceId[0]) {
+        this.selectedAudioInputDevice = deviceId[0];
       }
-
       this.selectedAudioInputTransformDevice = device;
     }
     this.publishSelectedAudioInputDevice();
     this.publishSelectAudioInputDeviceError();
     this.publishSelectedAudioInputTransformDevice();
   };
 
-  private getDeviceID = (
```

Comment:
Maybe I missed it but did you commit the changes in `device-utils.ts`?

Reply:
No, I replaced this with the `getIntrisicDeviceId` from the js sdk.